### ISSUE TITLE
Cannot import BadRequestError from lib/errors

### ISSUE THREAD

**Issue by okomarov** [gadicc/node-yahoo-finance2#202]
# Bug Report

## Describe the bug
Can't import specific errors from the lib/errors

## Minimal Reproduction
```
import { BadRequestError } from 'yahoo-finance2/dist/esm/src/lib/errors';

try {
  throw Error('Something');
} catch (e) {
  if (e instanceof BadRequestError) {
    console.log('aloha');
  }
  throw e;
}
```
then running it with:
```
yarn ts-node-dev src/test.ts
```
throws:
> Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './dist/esm/src/lib/errors' is not defined by "exports" in /Users/okomarov/Documents/repos/myrepo/api/node_modules/yahoo-finance2/package.json
    at throwExportsNotFound (internal/modules/esm/resolve.js:290:9)
    at packageExportsResolve (internal/modules/esm/resolve.js:513:3)
    at resolveExports (internal/modules/cjs/loader.js:437:36)
    at Function.Module._findPath (internal/modules/cjs/loader.js:477:31)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:872:27)
    at Function.Module._load (internal/modules/cjs/loader.js:730:27)
    at Module.require (internal/modules/cjs/loader.js:957:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/Users/okomarov/Documents/repos/myrepo/api/src/test.ts:1:1)
    at Module._compile (internal/modules/cjs/loader.js:1068:30)

## Environment
Node version (if applicable):  14.17.0
yarn version: 1.22.10
Library version (e.g. `1.10.1`): 1.12.1

**Comment by PythonCreator27** [comment]
@okomarov This isn't a bug. The error is just node saying that you can't import it, since the `exports` field prohibits it. This is 100% intended behavior.

@gadicc Is it possible to make the errors public? This pattern does seem to be [what MDN recommends](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch#conditional_catch-blocks).

**Comment by okomarov** [comment]
> @okomarov This isn't a bug. The error is just node saying that you can't import it, since the `exports` field prohibits it. This is 100% intended behavior.
> 
> @gadicc Is it possible to make the errors public? This pattern does seem to be [what MDN recommends](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch#conditional_catch-blocks).

Thanks for the quick reply! Feel free to change bug -> feature request or close.

**Comment by PythonCreator27** [comment]
I see the issue now. You could most likely switch over the `name` property (I really haven't tried it out). The `name` of a `BadRequestError` would be `"BadRequestError"`.

```ts
import yf from 'yahoo-finance2';
try {
    await yf.quote('AAPL');
} catch (err) {
    if (err.name === "BadRequestError") {
        // do something
    }
    // Check for more things...
}
```

Again, this code has not been tested, so no guarantees.

**Comment by gadicc** [comment]
Thanks @okomarov for reporting and yes, @PythonCreator27, I agree that the errors should be public.  Watch this space.

**Comment by gadicc** [comment]
I exposed the errors in `yahooFinance.errors`, and improved the [error handling docs](https://github.com/gadicc/node-yahoo-finance2/blob/devel/docs/README.md#error-handling).

You're welcome to check this out in a local `devel` copy but unless anything else comes up, I'll release this pretty soon.

**Comment by gadicc** [comment]
:tada: This feature is included in version 1.13.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/yahoo-finance2/v/1.13.0)
- [GitHub release](https://github.com/gadicc/node-yahoo-finance2/releases/tag/v1.13.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

**Comment by gadicc** [comment]
Thanks, @okomarov! :pray:
